### PR TITLE
Homework 4 completed

### PR DESCRIPTION
**Pull request by niavivek**

### CODE REVIEW

File: D04_ex00.py [hunk 1]
```diff
@@ -11,18 +11,47 @@
...
 ################################################################################
 # Imports
-
+import random
 
 # Body
 
+def guess_rand():
+	#get a random integer
+	x = random.randint(1,25)
```
Comment (danielsgriffin) [inline D04_ex00.py]: please add space: random.randint(1, 25)

File: D04_ex00.py [hunk 2]
```diff
@@ -11,18 +11,47 @@
...
+	#initialize the guess
+	y = ''
+	#counter for guesses
+	n = 1
+	#while guess is less than 6 and while guess is not same as random int
+	while y != x and n < 6:
```
Comment (danielsgriffin) [inline D04_ex00.py]: These variables are tough to understand.

File: HW04_ch07_ex02.py [hunk 3]
```diff
@@ -21,15 +21,27 @@
 
 ###############################################################################
 # Imports
-
+import math
 
 # Body
+def eval_loop():
```
Comment (danielsgriffin) [inline HW04_ch07_ex02.py]: "continue until the user enters 'done', and then **return** the value of the last expression it evaluated"

Comment (danielsgriffin) [inline HW04_ch07_ex02.py]: "continue until the user enters 'done', and then **return the value of the last expression it evaluated**"

File: HW04_ch08_ex04.py [hunk 4]
```diff
@@ -16,6 +16,7 @@
 def any_lowercase1(s):
     """Explain what is wrong, if anything, here.
     """
+    #this is correct, it returns True if it finds a lower case character
```
Comment (danielsgriffin) [inline HW04_ch08_ex04.py]: any_lowercase1("Daniel")

File: HW04_ch08_ex05.py [hunk 5]
```diff
@@ -2,3 +2,30 @@
 # See Chapter 8: Strings Exercise 5 for the task.
 # Please do provide function calls that test/demonstrate your
 # function.
+
+#function to perform caesar cypher
+def rotate_word(s,n):
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: Good job.

File: HW04_ch08_ex05.py [hunk 6]
```diff
@@ -2,3 +2,30 @@
...
+    #initialize rotated word
+    rot_string = ''
+    val = 0 # value to store starting point
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: Is this variable name informative?

**Comment by danielsgriffin** [comment]
Good job.

File: HW04_string2.py [hunk 7]
```diff
@@ -43,7 +59,21 @@ def not_bad(s):
...
+        else: # if b is odd, add the extra word to the first part
+             string_val = a[:int(half_a)] + b[:int(half_b)+1] + a[int(half_a):] + b[int(half_b)+1:]
+    elif half_b == int(half_b): # if only b is even, add the extra letter to first part of a
+        string_val = a[:int(half_a)+1] + b[:int(half_b)] + a[int(half_a)+1:] + b[int(half_b):]
+    else:#if both are odd, add the extra letter to first part of a and b
+        string_val = a[:int(half_a)+1] + b[:int(half_b)+1] + a[int(half_a)+1:] + b[int(half_b)+1:]
+    return string_val
 
 
 # Simple provided test() function used in main() to print
```
Comment (rohitraghavan) [inline HW04_string2.py]: Looks good!

**Pull request closed without merging**